### PR TITLE
[RFR] Add ability to traverse properties to ipappliance script

### PR DESCRIPTION
**Pull request by jkrocil**
Tested locally

Needed by rhev team to be able to check if DB has been enabled on an appliance, from jenkins.

`./scripts/ipappliance db.is_enabled <appliance_ip>`

### CODE REVIEW

File: scripts/ipappliance.py [hunk 1]
```diff
@@ -46,13 +50,26 @@ def log_callback(s):
...
 
     try:
-        call = getattr(appliance, action)
+        call = getattr(target_obj, action)
     except AttributeError:
         raise Exception('Action "{}" not found'.format(action))
-    if isinstance(getattr(type(appliance), action), property):
+
+    # The final obj may or may not be a callable
+    if not callable(call):
```
Comment (jkrocil) [inline scripts/ipappliance.py]: @mfalesni This ok?

File: scripts/ipappliance.py [hunk 2]
```diff
@@ -46,13 +50,26 @@ def log_callback(s):
 def call_appliance(ip_address, action, args, kwargs):
     # Given a provider class, find the named method and call it with
     # *args. This could possibly be generalized for other CLI tools.
-    appliance = IPAppliance(ip_address)
+    target_obj = IPAppliance(ip_address)
+
+    # Iterate over non-callables, such as appliance.db
+    try:
+        while '.' in action:
```
Comment (mfalesni) [inline scripts/ipappliance.py]: `for field in action.split('.')[:-1]:`

File: scripts/ipappliance.py [hunk 3]
```diff
@@ -46,13 +50,26 @@ def log_callback(s):
...
+            target_obj = getattr(target_obj, action.split('.', 1)[0])
+            if not callable(target_obj):
+                action = action.split('.', 1)[1]
+            else:
+                raise AttributeError
```
Comment (mfalesni) [inline scripts/ipappliance.py]: `raise AttributeError('field {} is not callable'.format(field))`

File: scripts/ipappliance.py (hunk 1 above)
Comment (mfalesni) [reply scripts/ipappliance.py]: I guess so.

**Comment by jkrocil** [comment]
Ok; hows this?
```
$ ./scripts/ipappliance.py is_web_ui_running --kwargs '{"unsure":false}' 10.8.199.234 2>&1 | tail -1
True
$ ./scripts/ipappliance.py db.is_enabled 10.8.199.234 2>&1 | tail -1
True
$ ./scripts/ipappliance.py address 10.8.199.234 2>&1 | tail -1
10.8.199.234
$ ./scripts/ipappliance.py bla 10.8.199.234 2>&1 | tail -1
The call on the appliance 10.8.199.234 failed: Action "bla" not found
$ ./scripts/ipappliance.py bla.bla 10.8.199.234 2>&1 | tail -1
The call on the appliance 10.8.199.234 failed: Field "bla" not found for object "IPAppliance(address='10.8.199.234', container=None, db_host=None, db_port=5432, ssh_port=22)"
```

**Pull request merged**